outbound.go

```diff
@@ -332,13 +307,22 @@ func cloneTags(tags map[string]string) map[string]string {
 // doneReading shuts down the message exchange for this call.
 // For outgoing calls, the last message is reading the call response.
 func (response *OutboundCallResponse) doneReading(unexpected error) {
-	now := response.GetTime()
-	response.AddAnnotationAt(AnnotationKeyClientReceive, now)
-	response.Report()
+	now := response.timeNow()
 
 	isSuccess := unexpected == nil && !response.ApplicationError()
 	lastAttempt := isSuccess || !response.requestState.HasRetries(unexpected)
 
+	// TODO how should this work with retries?
```

Comment:
If no objections, I would like to leave this for another PR, this one is already huge and been outstanding for almost 3 weeks. The current logic replicates the prior logic where a span was reported on every retry, so no worse than before.